File: app/src/main/java/sk/company/androidboilerplate/injection/module/ActivityModule.java
package sk.company.androidboilerplate.injection.module;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;
import sk.company.androidboilerplate.injection.ActivityContext;

@Module
public class ActivityModule {

    private Activity mActivity;

    public ActivityModule(Activity activity) {
        mActivity = activity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }

    @Provides
    @ActivityContext
    Context providesContext() {
        return mActivity;
    }
}

File: app/src/main/kotlin/sk/company/androidboilerplate/injection/module/ActivityModule.kt
package sk.company.androidboilerplate.injection.module

import android.app.Activity
import android.content.Context
import dagger.Module
import dagger.Provides
import sk.company.androidboilerplate.injection.ActivityContext

@Module
class ActivityModule(private val activity: Activity) {

    @Provides
    fun provideActivity(): Context {
        return activity
    }
    @Provides
    @ActivityContext
    fun providesContext(): Context {
        return activity
    }



}